Intersect.Server.Core/Entities/Projectile.cs

```diff
@@ -305,9 +333,8 @@ public bool CheckForCollision(ProjectileSpawn spawn)
                     !spawn.Parent.HasGrappled &&
                     (spawn.X != Owner.X || spawn.Y != Owner.Y))
                 {
-                    if (!spawn.ProjectileBase.HomingBehavior && !spawn.ProjectileBase.DirectShotBehavior && 
-                        (spawn.Dir <= Direction.Right || spawn.Dir != Direction.None && Options.Instance.MapOpts.EnableDiagonalMovement)
-                    )
+                    if (!spawn.ProjectileBase.HomingBehavior && !spawn.ProjectileBase.DirectShotBehavior &&
```

Comment:
add a comment here, something like "grapple hooks are only allowed in the default projectile behavior"